scipy/stats/_stats_py.py

```diff
@@ -1123,7 +1124,8 @@ def _moment(a, moment, axis, *, mean=None):
             n_list.append(current_n)
 
         # Starting point for exponentiation by squares
-        mean = a.mean(axis, keepdims=True) if mean is None else mean
+        mean = (a.mean(axis, keepdims=True) if mean is None
+                else dtype(mean))
```

Comment:
The problem was that if `a` and `mean` were both integers, then `np.finfo` fails below. The solution is to ensure that `mean` is the `dtype` determined above.